kernel/aster-nix/src/fs/inode_handle/dyn_cap.rs

```diff
@@ -61,6 +64,67 @@ impl InodeHandle<Rights> {
         }
         self.0.readdir(visitor)
     }
+
+    pub fn position(&self) -> Result<i64> {
+        Ok(self.0.offset() as i64)
+    }
+
+    pub fn test_advisory_lock(&self, lock: &mut RangeLock) -> Result<()> {
+        let extension = match self.dentry().inode().extension() {
+            Some(extension) => extension,
+            None => {
+                warn!("Inode extension is not supportted, the lock could be placed");
+                lock.set_type(RangeLockType::F_UNLCK);
+                return Ok(());
+            }
+        };
+
+        match extension.get::<RangeLockList>() {
+            None => {
+                // The advisory lock could be placed if there is no lock list
+                lock.set_type(RangeLockType::F_UNLCK);
+            }
+            Some(range_lock_list) => {
+                range_lock_list.test_lock(lock);
+            }
+        }
+        Ok(())
+    }
+
+    pub fn set_advisory_lock(&self, lock: &RangeLock, is_nonblocking: bool) -> Result<()> {
```

Comment:
I prefer to use `set_range_lock` here

Reply:
Thank you for your suggestions. I will rename them.